src/firefly/js/charts/dataTypes/XYColsCDT.js

```diff
@@ -197,32 +197,54 @@ export function setZoom(chartId, chartDataElementId, selection=undefined) {
     }
 }
 
-function serverParamsChanged(oldParams, newParams) {
+function serverParamsChanged(oldParams, newParams, chartDataElement) {
     if (oldParams === newParams) { return false; }
     if (!oldParams || !newParams) { return true; }
 
-    const newServerParams = getServerCallParameters(newParams);
-    const oldServerParams = getServerCallParameters(oldParams);
-    return newServerParams.some((p, i) => {
-        return p !== oldServerParams[i];
-    });
+    const {tblId, data} = chartDataElement;
+
+    if (isLargeTable(tblId)) {
+        const newServerParams = getServerCallParameters(newParams);
+        const oldServerParams = getServerCallParameters(oldParams);
+        return newServerParams.some((p, i) => {
+            return p !== oldServerParams[i];
+        });
+    } else {
+        // 'x', 'y', 'sortBy', 'xErr', 'xErrLow', 'xErrHigh', 'yErr', 'yErrLow', 'yErrHigh'
+        // server call parameters are present in the data
+        const newOpts = omitBy({
+            sortBy: newParams.sortColOrExpr,
+            x: newParams.x.columnOrExpr,
+            xErr: newParams.x.error,
+            xErrLow: newParams.x.errorLow,
+            xErrHigh: newParams.x.errorHigh,
+            y: newParams.y.columnOrExpr,
+            yErr: newParams.y.error,
+            yErrLow: newParams.y.errorLow,
+            yErrHigh: newParams.y.errorHigh
+        }, isUndefined);
+        return Object.keys(newOpts).some((o) => {
+            return newOpts[o] !== data[o];
+        });
+    }
 }
 
-function errorsRequested(xyPlotParams) {
-    return xyPlotParams &&
-        (xyPlotParams.x.error || xyPlotParams.y.error ||
-        xyPlotParams.x.errorLow || xyPlotParams.y.errorLow ||
-        xyPlotParams.x.errorHigh || xyPlotParams.y.errorHigh);
+function isLargeTable(tblId) {
+    const {totalRows}= getTblById(tblId);
+    return (totalRows >= 5000);
```

Comment:
I think it would make sense to convert the `5000` value into a constant, what do you think?